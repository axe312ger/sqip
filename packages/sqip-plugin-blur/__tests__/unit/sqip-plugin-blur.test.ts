import SvgPlugin from '../../src/sqip-plugin-blur'

const mockedConfig = {
  input: 'mocked',
  output: 'mocked',
  plugins: ['blur']
}

const sampleWithGroup =
  '<svg viewBox="0 0 1024 768"><rect fill="#bada55"/><g><path fill="#C0FFEE" d="M51.5 17.5l4 18 15 1z"/></g></svg>'
const sampleWithoutGroup =
  '<svg viewBox="0 0 1024 768"><rect fill="#bada55"/><polygon points="0,100 50,25 50,75 100,0" /></svg>'

describe('applies blur filter', () => {
  test('do nothing when no blur is given', () => {
    const svgPlugin = new SvgPlugin({
      options: {},
      sqipConfig: mockedConfig,
      pluginOptions: {
        blur: 0
      }
    })
    const result = svgPlugin.applyBlurFilter(sampleWithGroup)
    expect(result).toMatchSnapshot()
  })
  test('svg with group and blur', () => {
    const svgPlugin = new SvgPlugin({
      options: {},
      sqipConfig: mockedConfig,
      pluginOptions: {
        blur: 5
      }
    })
    const result = svgPlugin.applyBlurFilter(sampleWithGroup)
    expect(result).toMatchSnapshot()
  })
  test('svg without group and blur', () => {
    const svgPlugin = new SvgPlugin({
      options: {},
      sqipConfig: mockedConfig,
      pluginOptions: {
        blur: 5
      }
    })
    const result = svgPlugin.applyBlurFilter(sampleWithoutGroup)
    expect(result).toMatchSnapshot()
  })
})
