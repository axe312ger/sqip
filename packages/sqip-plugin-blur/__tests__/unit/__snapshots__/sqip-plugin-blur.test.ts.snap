// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applies blur filter do nothing when no blur is given 1`] = `"<svg viewBox=\\"0 0 1024 768\\"><rect fill=\\"#bada55\\"/><g><path fill=\\"#C0FFEE\\" d=\\"M51.5 17.5l4 18 15 1z\\"/></g></svg>"`;

exports[`applies blur filter svg with group and blur 1`] = `
"<svg viewBox=\\"0 0 1024 768\\"><g filter=\\"url(#b)\\"><rect fill=\\"#bada55\\"></rect><g><path fill=\\"#C0FFEE\\" d=\\"M51.5 17.5l4 18 15 1z\\"></path></g></g><filter id=\\"b\\">
        <feGaussianBlur in=\\"SourceGraphic\\" stdDeviation=\\"5\\"></feGaussianBlur>
        <feComponentTransfer>
            <feFuncA type=\\"discrete\\" tableValues=\\"1 1\\"></feFuncA>
        </feComponentTransfer>
      </filter></svg>"
`;

exports[`applies blur filter svg without group and blur 1`] = `
"<svg viewBox=\\"0 0 1024 768\\"><g filter=\\"url(#b)\\"><rect fill=\\"#bada55\\"></rect><polygon points=\\"0,100 50,25 50,75 100,0\\"></polygon></g><filter id=\\"b\\">
        <feGaussianBlur in=\\"SourceGraphic\\" stdDeviation=\\"5\\"></feGaussianBlur>
        <feComponentTransfer>
            <feFuncA type=\\"discrete\\" tableValues=\\"1 1\\"></feFuncA>
        </feComponentTransfer>
      </filter></svg>"
`;
